### DISCUSSION TITLE
Audit log crashes because too much data

### DISCUSSION THREAD

**Discussion by V0lantis** [apache/airflow#25638] [General]
### Apache Airflow version

2.3.3

### What happened

The audit log link crashes because there is too much data displayed.

### What you think should happen instead

The windows shouldn't crash

### How to reproduce

Display a dag audit log with thousand or millions lines should do the trick

### Operating System
```
NAME="Amazon Linux" VERSION="2" ID="amzn" ID_LIKE="centos rhel fedora" VERSION_ID="2" PRETTY_NAME="Amazon Linux 2" ANSI_COLOR="0;33" CPE_NAME="cpe:2.3:o:amazon:amazon_linux:2" HOME_URL="https://amazonlinux.com/"
```
### Versions of Apache Airflow Providers

```
apache-airflow-providers-amazon==4.0.0
apache-airflow-providers-celery==3.0.0
apache-airflow-providers-cncf-kubernetes==4.1.0
apache-airflow-providers-datadog==3.0.0
apache-airflow-providers-docker==3.0.0
apache-airflow-providers-ftp==3.0.0
apache-airflow-providers-github==2.0.0
apache-airflow-providers-google==8.1.0
apache-airflow-providers-grpc==3.0.0
apache-airflow-providers-hashicorp==3.0.0
apache-airflow-providers-http==3.0.0
apache-airflow-providers-imap==3.0.0
apache-airflow-providers-jira==3.0.0
apache-airflow-providers-mysql==3.0.0
apache-airflow-providers-postgres==5.0.0
apache-airflow-providers-redis==3.0.0
apache-airflow-providers-sftp==3.0.0
apache-airflow-providers-slack==5.0.0
apache-airflow-providers-sqlite==3.0.0
apache-airflow-providers-ssh==3.0.0
apache-airflow-providers-tableau==3.0.0
apache-airflow-providers-zendesk==4.0.0
```

### Deployment

Official Apache Airflow Helm Chart

### Deployment details

k8s

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by uranusjr** [comment]
Does the web server crash, or the browser window crash? I don’t think we can do too much if it’s the browser window, maybe except changing the page to download if the log is over some certain threshold. But this is very arbitrary since the browser is out of Airflow’s control.

**Comment by potiuk** [comment]
Yeah. we need to know the answer what crashed. If your browser is configured to open and render ".log" files inline, we can add download attribute to re https://www.w3schools.com/howto/howto_html_download_link.asp (until then you can right-click and choose download). Please let us know if we figured it correctly @V0lantis

**Comment by V0lantis** [comment]
Thanks both for your answers. It is indeed the web browser which crashes 
![image](https://user-images.githubusercontent.com/37664438/183911989-3bd56107-a5d6-4cf3-9872-9162733f1b89.png)
The `save link` indeed works, but it saves the link into html. I admit it would beneficiate from a nicer format (like `.log` extension with just the basic logs?)

*Edit:*
When I tried to just open the html file which resulted from `save as` link, the page was weighting 2go in the chrome task manager tool

**Reply by V0lantis** [reply]
Also something else, when opening the basic `audit_log.html` file, the logs are not sorted by dates, which don't help understanding the flow of things which happened. I don't know if this the case when we open the audit log directly from the link (and that the webrowser doesn't crash)

**Reply by V0lantis** [reply]
Sorry, I am a little bit spamming here.
In a bare new instance, I can display the Audit log, and this is indeed very interesting ! You can also sort everything as you will. A shame that you cannot do that when the audit logs are too big

**Reply by potiuk** [reply]
I created issue back from it: https://github.com/apache/airflow/issues/25641 . and marked it as good-first-issue, maybe someone would like to pick it up. But if you want to make it happen quickly you are most welcome to contribute a PR back fixing it. Airflow has > 2100 contributors and many of them are users like you, who contributed back something that annoyed them. This is how FOSS software works.

**Reply by V0lantis** [reply]
Absolutely, thank-you @potiuk

**Comment by AndrewLaganaro** [comment]
Same here. the log has ~70MB of data and when clicking the green square of the task, the whole browser tab freezes forever
I have 32GB of ram and I'm not even close to fill it up, so I think its a problem with airflow webserver

Im using `Opera One(ver: 109.0.5097.68)`
